test/cmd/node-management.sh

```diff
@@ -121,15 +135,17 @@ __EOF__
   # Pre-condition: test-pod-1 and test-pod-2 exist
   kube::test::get_object_assert "pods" "{{range .items}}{{.metadata.name}},{{end}}" 'test-pod-1,test-pod-2,'
   # dry-run command
-  kubectl drain "127.0.0.1" --pod-selector 'e in (f)' --dry-run=client
-  kubectl drain "127.0.0.1" --pod-selector 'e in (f)' --dry-run=server
+  kubectl drain "127.0.0.1" --pod-selector 'e in (f)' --dry-run=client --force
+  kubectl drain "127.0.0.1" --pod-selector 'e in (f)' --dry-run=server --force
   kube::test::get_object_assert "pods" "{{range .items}}{{.metadata.name}},{{end}}" 'test-pod-1,test-pod-2,'
-  # command
-  kubectl drain "127.0.0.1" --pod-selector 'e in (f)'
-  # only "test-pod-1" should have been matched and deleted - test-pod-2 should still exist
-  kube::test::get_object_assert "pods/test-pod-2" "{{.metadata.name}}" 'test-pod-2'
-  # delete pod no longer in use
-  kubectl delete pod/test-pod-2
+  # command - need --force because pod is unmanaged and --skip-wait-for-delete-timeout because node is unready
+  response=$(kubectl drain "127.0.0.1" --force --pod-selector 'e in (f)' --skip-wait-for-delete-timeout=1)
+  kube::test::if_has_string "${response}" "evicting pod .*/test-pod-1"
+  # only "test-pod-1" should have been matched and deleted - test-pod-2 should not have a deletion timestamp
+  kube::test::get_object_assert "pods/test-pod-2" "{{.metadata.deletionTimestamp}}" '<no value>'
+  # Post-condition: recreate test pods -- they have deletionTimestamp set but will not go away because node is unready
+  delete_test_pods
+  create_test_pods
```

Comment:
Recreating because I'll need them again in the tests I'm adding in https://github.com/kubernetes/kubernetes/pull/100148